src/inference/dev_api/cpp_interfaces/interface/ie_iinfer_request_internal.hpp

```diff
@@ -245,6 +273,24 @@ class INFERENCE_ENGINE_API_CLASS(IInferRequestInternal) : public std::enable_sha
 
     void addInputPreProcessingFor(const std::string& name, const Blob::Ptr& from, const Blob::Ptr& to);
 
+    /**
+     * @brief Performs actual concatenation of blobs into single tensor
+     * Default implementation may allocate memory for new blob containing user's input data
+     * Plugin is allowed to override this behavior
+     * @throws Exception if error occurs
+     */
+    virtual void convertBatchedBlob(const std::string& name, const InferenceEngine::BatchedBlob::Ptr& blob);
+
+    /**
+     * @brief Performs basic validation of user's blobs set via ov::runtime::InferRequest
+     * 'set_tensors/set_input_tensors'
+     * @note Plugin-specific implementations may call this function to performs basic validation inside 'SetBlobs'
+     * @param name - input name.
+     * @param blobs - input blobs. The type of Blob must correspond to the network input
+     * precision and size.
+     */
+    virtual void checkBatchedBlobs(const std::string& name, const std::vector<Blob::Ptr>& blobs);
```

Comment:
the name is a bit misleading, as the above func accepts the BatchedBlob::Ptr, then the checkBatchedBlob**s** assumingly should accept vector of the BatchedBlob::Ptr... maybe checkBlobsForBatch?

Reply:
BTW, maybe `convertBatchedInputBlobs`?

Reply:
renamed to checkBlobsForBatch